sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -531,6 +532,22 @@ static int sof_card_late_probe(struct snd_soc_card *card)
 	return hdac_hdmi_jack_port_init(component, &card->dapm);
 }
 
+static int soc_card_add_dai_link(struct snd_soc_card *card,
+				 struct snd_soc_dai_link *dai_link)
+{
+	if (!strcmp(dai_link->name, "NHLT-RENDER-CAPTURE") ||
+	    !strcmp(dai_link->name, "NHLT-RENDER-ONLY") ||
+	    !strcmp(dai_link->name, "NHLT-CAPTURE-ONLY") ||
+	    !strcmp(dai_link->name, "NHLT-BT")) {
+		/* fixup and register this dai link later when we receive NHLT
+		 * table from IPC4 topology
+		 */
+		dai_link->ignore = true;
```

Comment:
I don't know what this 'ignore' means but it certainly does not immediately translate as 'fixup and register later'.

Reply:
soc-core uses this flag to skip FE dai links in soc_check_tplg_fes() function. Here we borrow this flag to make snd_soc_add_pcm_runtime() function skipping this dai link.

Reply:
"borrow this flag" really means overloading a concept used for FE and extend it for BEs. That's not so good, this will add confusions left and right. We should be really careful with such moves.

Reply:
I checked the soc.h and the flag means "do not create pcm". Maybe the reason to use this flag could be different? soc-core uses this flag to skip FE links and we use it to skip links which link/stream names need to be resolved later.

	/* Do not create a PCM for this DAI link (Backend link) */
	unsigned int ignore:1;

Reply:
It's best to avoid using the same flag for two completely different purposes, which means adding a new one with a clearly defined behavior.

I am still in the dark on what the problem statement is and why the SOF driver is involved.

Reply:
I agree we cannot repurpose an existing flag for a compltely different purpose